erpnext/stock/doctype/quality_inspection/quality_inspection.json

```diff
@@ -226,12 +244,99 @@
    "label": "Status",
    "options": "\nUntested\nAccepted\nRejected\nPartially Rejected\nSkipped",
    "reqd": 1
+  },
+  {
+   "fieldname": "section_break_strain_test",
+   "fieldtype": "Section Break",
+   "label": "Strain Test"
+  },
+  {
+   "fieldname": "strain_notes",
+   "fieldtype": "Text",
+   "label": "Strain Notes"
+  },
+  {
+   "fieldname": "testing_lab",
+   "fieldtype": "Link",
+   "label": "Testing Lab",
+   "options": "Supplier"
+  },
+  {
+   "fieldname": "column_break_strain_test",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "testing_result_link",
+   "fieldtype": "Data",
+   "label": "Testing Result Link"
+  },
+  {
+   "fieldname": "test_date",
+   "fieldtype": "Date",
+   "label": "Test Date"
+  },
+  {
+   "fieldname": "section_break_manufacture_information",
+   "fieldtype": "Section Break",
+   "label": "Manufacture Information"
+  },
+  {
+   "fieldname": "manufacturer_name",
+   "fieldtype": "Link",
+   "label": "Manufacturer Name",
+   "options": "Supplier"
+  },
+  {
+   "fieldname": "column_break_manufacture_information",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "manufacturer_website",
+   "fieldtype": "Data",
+   "label": "Manufacturer Website"
+  },
+  {
+   "fieldname": "sign_off",
+   "fieldtype": "Signature",
+   "label": "Verifier\u2019s signature"
+  },
+  {
+   "fieldname": "secation_break_purchase_details",
+   "fieldtype": "Section Break",
+   "label": "Purchase Details"
+  },
+  {
+   "fieldname": "qty",
+   "fieldtype": "Data",
+   "label": "Accepted Quantity"
+  },
+  {
+   "fieldname": "column_break_purchase_details",
+   "fieldtype": "Column Break"
+  },
+  {
+   "fieldname": "uom",
+   "fieldtype": "Data",
+   "label": "UOM"
+  },
```

Comment:
These should be read-only and dependent on whether the QI is being made against a Purchase document, since they'll only be pulled from there.